operator/pkg/model/translation/gateway-api/translator.go

```diff
@@ -131,7 +131,7 @@ func getEndpoints(resource model.FullyQualifiedResource) *corev1.Endpoints {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      shorten(ciliumGatewayPrefix + resource.Name),
 			Namespace: resource.Namespace,
-			Labels:    map[string]string{owningGatewayLabel: resource.Name},
+			Labels:    map[string]string{owningGatewayLabel: shorten(resource.Name)},
```

Comment:
Do we need to record the original names in annotations?

Reply:
we will need to have similar change in the below function, as the label is used to fetch related service

https://github.com/cilium/cilium/blob/3f9e0e9c4ad5766eb2210d6aacf774745c694f24/operator/pkg/gateway-api/gateway_reconcile.go#L318

Reply:
> we might need to have similar change in the below function, as the label is used to fetch related service


Updated.